### PR TITLE
Language Retrieval

### PR DESCRIPTION
**Pull request by 16dprice**
This PR aims to define how the rest of the system should get available languages and the relevant information for those languages to be displayed to the user.

---
This change is [Reviewable](https://reviewable.io/reviews/bible-translation-tools/fetcher/11)

### CODE REVIEW

**Comment by AnonymousWalker** [comment]
___
*[src/main/kotlin/org/bibletranslationtools/fetcher/application/PortLanguageRepository.kt, line 10 at r1](https://reviewable.io/reviews/bible-translation-tools/fetcher/11#-MBFdL4h9O22xmdYm-MU:-MBFkYuS4Pj_wM_O2fvD:baz2u5) ([raw file](https://github.com/bible-translation-tools/fetcher/blob/40fc23aca83136c7a539b8d9f8a73ce47ac3a7dd/src/main/kotlin/org/bibletranslationtools/fetcher/application/PortLanguageRepository.kt#L10)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

how about we change this to
PortGatewayLanguageCatalog : LanguageCatalog

The SourceAudioLanguages class isn't well named (the name doesn't tell me anything meaningful, or worse, leads me to believe it's something different)

What you probably want there is to start constructing a more "true" "repository" where you get the content. It might have things like getLanguages(), getBooks(languageCode: String), etc...
</blockquote></details>

@jsarabia  do you mean that a repo that provides many functionalities like getLanguages(), getBooks().... and such?

**Comment by 16dprice** [comment]
___
*[src/main/kotlin/org/bibletranslationtools/fetcher/application/PortLanguageRepository.kt, line 36 at r1](https://reviewable.io/reviews/bible-translation-tools/fetcher/11#-MBFbRzG6RDLP19iDgSX:-MBFzMXq0Q1QZbSg6mTg:b-896fix) ([raw file](https://github.com/bible-translation-tools/fetcher/blob/40fc23aca83136c7a539b8d9f8a73ce47ac3a7dd/src/main/kotlin/org/bibletranslationtools/fetcher/application/PortLanguageRepository.kt#L36)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

I think you can just call getResource directly and don't need [this...]?
</blockquote></details>

Done.

File: src/main/kotlin/org/bibletranslationtools/fetcher/application/SourceAudioRepository.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package org.bibletranslationtools.fetcher.application
+
+import java.nio.file.Files
+import java.nio.file.Paths
+import kotlin.streams.toList
+import org.bibletranslationtools.fetcher.data.Language
+import org.bibletranslationtools.fetcher.domain.LanguageCatalog
+
+class SourceAudioRepository(
```
Comment (AnonymousWalker) [inline src/main/kotlin/org/bibletranslationtools/fetcher/application/SourceAudioRepository.kt:9]: I think this is a use case and its name should not be named "Repository".

https://medium.com/slalom-build/clean-architecture-with-java-11-f78bba431041
This is a good example I found that represented a use case with the repository passed into it.

Comment (AnonymousWalker) [reply src/main/kotlin/org/bibletranslationtools/fetcher/application/SourceAudioRepository.kt:9]: How about SourceAudioContent? We then can use: SourceAudioContent().getLanguages()

**Comment by AnonymousWalker** [comment]
___
*[src/main/kotlin/org/bibletranslationtools/fetcher/application/SourceAudioRepository.kt, line 9 at r5](https://reviewable.io/reviews/bible-translation-tools/fetcher/11#-MBFzN71z7IjKjRogHti-r5-9:-MBeif_78iAOsFh-r48M:bjiyq82) ([raw file](https://github.com/bible-translation-tools/fetcher/blob/7ff97d0c878f0e9b3711e564d46e883e663a3c43/src/main/kotlin/org/bibletranslationtools/fetcher/application/SourceAudioRepository.kt#L9)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

@anonymouswalker Is this an entity (noun) or an action (verb)?
</blockquote></details>

Oh I forgot that the use case should be verb

**Pull request merged**